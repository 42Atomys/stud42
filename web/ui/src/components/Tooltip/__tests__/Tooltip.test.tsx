import TooltipDefault, { Tooltip } from '@components/Tooltip';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

it('snapshot: renders Tooltip unchanged', () => {
  const { container } = render(
    <TooltipDefault text="tooltip-test">Test</TooltipDefault>
  );
  expect(container).toMatchSnapshot();
});

it('state: hover changes', async () => {
  render(<Tooltip text="test-tooltip">Test</Tooltip>);

  fireEvent.mouseEnter(screen.getByTestId('hover-target'));
  await waitFor(() => screen.getByTestId('tooltip'));

  expect(screen.getByTestId('tooltip')).toHaveClass('visible z-10');
  expect(screen.getByTestId('tooltip')).not.toHaveClass('invisible');

  fireEvent.mouseLeave(screen.getByTestId('hover-target'));
  await waitFor(() => screen.getByTestId('tooltip'));

  expect(screen.getByTestId('tooltip')).toHaveClass('invisible');
  expect(screen.getByTestId('tooltip')).not.toHaveClass('visible z-10');
});
