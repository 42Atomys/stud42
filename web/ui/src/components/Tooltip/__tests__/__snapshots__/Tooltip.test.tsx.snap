// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: renders Tooltip unchanged 1`] = `
<div>
  <div
    class="relative flex items-center"
    data-testid="hover-target"
  >
    <div
      class="hidden -z-10 opacity-0 ml-1 absolute left-full bg-gradient-to-l dark:bg-gradient-to-r from-indigo-400 dark:from-indigo-600 to-indigo-500 dark:to-indigo-500 text-slate-100 dark:text-slate-300 px-4 py-2 rounded transition-all duration-150 max-w-[250px] flex-col w-max items-start"
      data-testid="tooltip"
    >
      <div
        class="bg-indigo-500 dark:bg-indigo-600 absolute w-3 h-3 top-1/3 whitespace-nowrap"
        style="left: -6px; transform: rotate(45deg);"
      />
      <span
        class=""
      >
        tooltip-test
      </span>
    </div>
    Test
  </div>
</div>
`;
